src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/RcaController.java

```diff
@@ -255,6 +267,47 @@ private void readRcaEnabledFromConf() {
         });
   }
 
+  /**
+   * Reads the mutedRCAList value from the rca.conf file, performs validation on the param value
+   * provided and on successful validation, updates the AnalysisGraph with muted RCA value.
+   *
+   * <p>In case all the RCAs in param value are incorrect, return without any update.
+   */
+  private void readAndUpdateMutesRcas() {
+    // If the rca config file has been updated since the lastModifiedTimeInMillisInMemory in memory,
+    // refresh the `muted-rcas` value from rca config file.
+    final RcaConf rcaConf = RcaControllerHelper.pickRcaConfForRole(currentRole);
+    long lastModifiedTimeInMillisOnDisk = rcaConf.getLastModifiedTime();
+    if (lastModifiedTimeInMillisOnDisk > lastModifiedTimeInMillisInMemory) {
+      try {
+        List<String> rcasForMute = rcaConf.getMutedRcaList();
+        LOG.info("RCAs provided for muting : {}", rcasForMute);
+
+        Set<String> graphNodeNames = new HashSet<>();
+        RcaUtil.getAnalysisGraphComponents(rcaConf).forEach(
```

Comment:
We are iterating through the list of nodes three times: 1. in `getAllNodes()`, then again in `getNodeNames()` and then here. And we are create new HasSets everytime. Let's try to minimize the HashSet creation and iterations.

Reply:
Done, have updated the code to use a single hashmap, rather than creating multiple.